Deprecate WebCrypto RSAES-PKCS1-v1_5 support

youennf opened the pull request:
#### 2aee342cf4c4c0ef893c132314a2e529aa2c5335
<pre>
Deprecate WebCrypto RSAES-PKCS1-v1_5 support
<a href="https://bugs.webkit.org/show_bug.cgi?id=247358">https://bugs.webkit.org/show_bug.cgi?id=247358</a>
rdar://101765059

Reviewed by J Pascoe.

Add a runtime flag to enable/disable RSAES-PKCS1-v1_5 support.
Mark it as experimental. Update tests covering RSAES-PKCS1-v1_5 to enable this flag.
Based on the flag, all WebCrypto operations related to RSAES-PKCS1-v1_5 will either return NotSupported or proceeed as before.

* LayoutTests/crypto/subtle/derive-key-malformed-parameters-expected.txt:
* LayoutTests/crypto/subtle/generate-key-malformed-parameters-expected.txt:
* LayoutTests/crypto/subtle/import-key-malformed-parameters-expected.txt:
* LayoutTests/crypto/subtle/rsa-import-key-malformed-parameters-expected.txt:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-jwk.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-pkcs8.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-spki.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-encrypt-decrypt.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-extractable.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-private-key.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-empty-usages.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-leading-zero.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-minimum.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-non-extractable.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-decrypt.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-encrypt.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-pkcs8-key.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key-empty-usages.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key.html:
* LayoutTests/crypto/workers/subtle/rsa-postMessage-worker.html:
* LayoutTests/crypto/workers/subtle/rsaes-pkcs1-v1_5-import-key-decrypt.html:
* LayoutTests/crypto/workers/subtle/rsaes-pkcs1-v1_5-import-key-encrypt.html:
* LayoutTests/platform/glib/crypto/subtle/rsa-generate-key-malformed-parameters-expected.txt: Removed.
* LayoutTests/platform/glib/crypto/subtle/rsa-import-key-malformed-parameters-expected.txt: Removed.
* LayoutTests/platform/ios/crypto/subtle/rsa-generate-key-malformed-parameters-expected.txt: Removed.
* LayoutTests/platform/ios/crypto/subtle/rsa-import-key-malformed-parameters-expected.txt: Removed.
* LayoutTests/platform/mac/crypto/subtle/rsa-generate-key-malformed-parameters-expected.txt: Removed.
* LayoutTests/platform/mac/crypto/subtle/rsa-generate-key-malformed-parameters-expected.txt:
* LayoutTests/platform/mac/crypto/subtle/rsa-import-key-malformed-parameters-expected.txt: Removed.
* LayoutTests/platform/wincairo/crypto/subtle/rsa-generate-key-malformed-parameters-expected.txt: Removed.
* LayoutTests/platform/wincairo/crypto/subtle/rsa-import-key-malformed-parameters-expected.txt: Removed.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::isRSAESPKCSWebCryptoDeprecated):
(WebCore::normalizeCryptoAlgorithmParameters):
(WebCore::isSupportedExportKey):
(WebCore::SubtleCrypto::exportKey):
(WebCore::SubtleCrypto::wrapKey):

Canonical link: <a href="https://commits.webkit.org/259018@main">https://commits.webkit.org/259018@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/383f7c76ca17e1b42a244f9b6ae2779b4e4e6da0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102639 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11769 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35674 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/111906 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/172129 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12773 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2675 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/94914 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/109609 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/108415 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/9802 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/93021 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/37460 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/91654 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24529 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/79203 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/5225 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/25948 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5385 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/2396 "Found 1 new test failure: webgl/2.0.0/conformance2/textures/canvas_sub_rectangle/tex-3d-rgb8-rgb-unsigned_byte.html (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/11406 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45437 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6193 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/7115 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

pascoej reviewed (approved):
This LGTM. If anything was used from https://github.com/WebKit/WebKit/pull/6040, please add credit in the commit message.

youennf committed:
Deprecate WebCrypto RSAES-PKCS1-v1_5 support
https://bugs.webkit.org/show_bug.cgi?id=247358
rdar://101765059

Reviewed by J Pascoe.

Add a runtime flag to enable/disable RSAES-PKCS1-v1_5 support.
Mark it as experimental. Update tests covering RSAES-PKCS1-v1_5 to enable this flag.
Based on the flag, all WebCrypto operations related to RSAES-PKCS1-v1_5 will either return NotSupported or proceeed as before.

* LayoutTests/crypto/subtle/derive-key-malformed-parameters-expected.txt:
* LayoutTests/crypto/subtle/generate-key-malformed-parameters-expected.txt:
* LayoutTests/crypto/subtle/import-key-malformed-parameters-expected.txt:
* LayoutTests/crypto/subtle/rsa-import-key-malformed-parameters-expected.txt:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-jwk.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-pkcs8.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-export-key-spki.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-encrypt-decrypt.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key-extractable.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-generate-key.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-private-key.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-empty-usages.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-leading-zero.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-minimum.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key-non-extractable.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-jwk-public-key.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-decrypt.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-key-encrypt.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-pkcs8-key.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key-empty-usages.html:
* LayoutTests/crypto/subtle/rsaes-pkcs1-v1_5-import-spki-key.html:
* LayoutTests/crypto/workers/subtle/rsa-postMessage-worker.html:
* LayoutTests/crypto/workers/subtle/rsaes-pkcs1-v1_5-import-key-decrypt.html:
* LayoutTests/crypto/workers/subtle/rsaes-pkcs1-v1_5-import-key-encrypt.html:
* LayoutTests/platform/glib/crypto/subtle/rsa-generate-key-malformed-parameters-expected.txt: Removed.
* LayoutTests/platform/glib/crypto/subtle/rsa-import-key-malformed-parameters-expected.txt: Removed.
* LayoutTests/platform/ios/crypto/subtle/rsa-generate-key-malformed-parameters-expected.txt: Removed.
* LayoutTests/platform/ios/crypto/subtle/rsa-import-key-malformed-parameters-expected.txt: Removed.
* LayoutTests/platform/mac/crypto/subtle/rsa-generate-key-malformed-parameters-expected.txt: Removed.
* LayoutTests/platform/mac/crypto/subtle/rsa-generate-key-malformed-parameters-expected.txt:
* LayoutTests/platform/mac/crypto/subtle/rsa-import-key-malformed-parameters-expected.txt: Removed.
* LayoutTests/platform/wincairo/crypto/subtle/rsa-generate-key-malformed-parameters-expected.txt: Removed.
* LayoutTests/platform/wincairo/crypto/subtle/rsa-import-key-malformed-parameters-expected.txt: Removed.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::isRSAESPKCSWebCryptoDeprecated):
(WebCore::normalizeCryptoAlgorithmParameters):
(WebCore::isSupportedExportKey):
(WebCore::SubtleCrypto::exportKey):
(WebCore::SubtleCrypto::wrapKey):

Canonical link: https://commits.webkit.org/259018@main
